request.go

```diff
@@ -213,6 +213,22 @@ func (r Request) ResponseWithCode(body interface{}, statusCode int) Response {
 	return rsp
 }
 
+// RouterEndpointPattern finds the router pattern that matches the request. This is only callable while the request
+// is being served.
+func (r Request) RouterEndpointPattern() string {
+	if router := RouterForRequest(r); router != nil {
+		if pathPattern := router.Pattern(r); pathPattern != "" {
+			return pathPattern
+		}
+	}
```

Comment:
An alternative implementation here would be to store the pattern used to serve the request in the context, [as we do with the router currently](https://github.com/monzo/typhon/blob/ea17ae8b9d577dfbeeedf388993b22ba52867da5/router.go#L119-L126). Then we don't need to look up the pattern separately (which could theoretically have changed if a new route is registered in the intervening time).